client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/step-content.tsx

```diff
@@ -9,11 +12,32 @@ type StepContentProps = {
 	goToStep?: NavigationControls[ 'goToStep' ];
 };
 
-const StepContent = ( { siteSlug, submit, goNext, goToStep }: StepContentProps ) => (
-	<div className="launchpad__content">
-		<Sidebar siteSlug={ siteSlug } submit={ submit } goNext={ goNext } goToStep={ goToStep } />
-		<LaunchpadSitePreview siteSlug={ siteSlug } />
-	</div>
-);
+const StepContent = ( { siteSlug, submit, goNext, goToStep }: StepContentProps ) => {
+	const site = useSite();
+
+	const { data: allDomains = [] } = useGetDomainsQuery( site?.ID ?? null, {
+		retry: false,
+	} );
+
+	const [ freeDomain, setFreeDomain ] = useState< string | null >( null );
+
+	useEffect( () => {
+		if ( allDomains ) {
+			allDomains.forEach( ( domainObject ) => {
+				// find the free .wordpress.com domain which will always be secure (will have SSL certificate)
+				if ( domainObject.domain.endsWith( '.wordpress.com' ) ) {
+					setFreeDomain( domainObject.domain );
+				}
+			} );
```

Comment:
Is there some sort of safety fallback we can add if nothing is found in the `forEach` above?
```
if ( allDomains ) {
    *.forEach ( ....
           if ( domainObject.endsWith( 'wordpress.com') {  setFreeDomain(..) }
      )
      if ( ! freeDomain ) { setFreeDomain( somethingElse? / Primary domain? ) } 
}
```      

I would imagine alternatively we could normally set a default value for `freeDomain` in the `useState` setup above, but that seems like it could be problematic given our original problem - potentially still reference a non-secure custom primary in the iframe  before `allDomains` from the useEffect runs -> breaking the launchpad.